edxlearndot/learndot.py

```diff
@@ -377,8 +377,9 @@ def set_enrolment_status(self, enrolment_id, status, unconditional=False):
                 Requests throws anything.
         """
         if not enrolment_id:
-            log.error("Enrolment_id can not be None")
-            return
+            msg = "Enrolment_id can not be None"
+            log.error(msg)
+            raise LearndotAPIException(msg)
```

Comment:
Hmm.. only problem with this is, now we get an exception thrown in the [signal handler](https://github.com/open-craft/edx-learndot/blob/ca6a4fec17642e45b01aeb251877aef70c1d98f1/edxlearndot/signals.py#L46) and the [management command](https://github.com/open-craft/edx-learndot/blob/ca6a4fec17642e45b01aeb251877aef70c1d98f1/edxlearndot/management/commands/update_learndot_enrolments.py#L109) when it encounters a user with no enrolments.

Better to just return None maybe?

Reply:
Or you could combine all three of these calls (`get_contact_id`, `get_enrolment_id`, and `set_enrolment_status_to_passed`) into a single method that takes the `User` and the `CourseMapping`, and does the right thing?

Reply:
I've added the new method as you suggested, but it's only a combination of get_enrolment_id and set_enrolment_status_to_passed, because contact_id should be calculated before doing the call, otherwise we will be duplicating calls on the signal handler, see: https://github.com/open-craft/edx-learndot/blob/ca6a4fec17642e45b01aeb251877aef70c1d98f1/edxlearndot/signals.py#L41